### ISSUE TITLE
Do not mangle classes names in WebApi Model

### ISSUE THREAD

**Issue by postatum** [raml-org/webapi-parser#9]
Find a way to not mangle names of classes when building JS from Scala with `fullOptJS`.
Also note how `.js` file size changes.

This should make easier for users to explore/navigate AMF model.

Ideal solution should:
1. Leave names exported from Scala to JS un-mangled. At least the part which is displayed when logging webapi-parser Model objects in NodeJS console.
2. Mangle/minimize everything else as much as possible (what fullOptJS does).
3. Output JS size should be affected as little as possible.
4. Still produce working JS and Java packages.

**Comment by artemkaint** [comment]
+1 It would be great

**Comment by yungcheng** [comment]
It seems you just need to provide a build time configuration to do `fastOptJS` instead of `fullOptJS`. Previously the webapi-parser was using `fastOptJS`
https://github.com/raml-org/webapi-parser/commit/efe220d502bb81122de5ea7137cf6cba1f301dec

**Comment by postatum** [comment]
> It seems you just need to provide a build time configuration to do `fastOptJS` instead of `fullOptJS`. Prior to this commit [efe220d](https://github.com/raml-org/webapi-parser/commit/efe220d502bb81122de5ea7137cf6cba1f301dec), the webapi-parser was using `fastOptJS`

Hi @yungcheng. Thank you for helping us address this issue!
Though, simply switching `fullOptJS` to `fastOptJS` doesn't fully fix the issue, because `fastOptJS` adds too much weight to the output JS file (it explodes to 20.1mb from 3-4mb).

Ideal solution should:
1. Leave exported Scala names un-mangled. At least the part which is displayed when logging webapi-parser Model objects in NodeJS console.
2. Mangle/minimize everything else as much as possible (what fullOptJS does).
3. Still produce a working JS and Java packages.

**Comment by yungcheng** [comment]
Scala js uses Google Closure Compiler, in order to do full optimization but keep the function name intact, you’d need to have the ability to configure how closure compiler is run. Also if I understand correctly, what you want is not an artifact for development time but a prod artifact with functions unobfuscated?

**Comment by postatum** [comment]
> Scala js uses Google Closure Compiler, in order to do full optimization but keep the function name intact, you’d need to have the ability to configure how closure compiler is run. Also if I understand correctly, what you want is not an artifact for development time but a prod artifact with functions unobfuscated?

Correct. I want WebApi Model functions and properties names unobfuscated when logging object in a Nodejs console.

**Comment by mmmveggies** [comment]
This library feels unusable. I can't inspect the output of the library and the [type definitions](https://github.com/raml-org/webapi-parser/blob/master/js/module/typings/amf-client-js.d.ts#L594) don't match [the example](https://github.com/raml-org/webapi-parser/blob/master/examples/js/raml10-model-navigation.js).

How are people browsing their RAML models like this? Are there just known/expected properties?
